src/types.d.ts

```diff
@@ -11,4 +12,8 @@ declare global {
 			};
 		};
 	}
+
+	interface HTMLElementTagNameMap {
+		"purple-dot-checkout": PurpleDotCheckoutElement;
+	}
```

Comment:
This is my preferred way to globally handle types for web components, it saves us from having to do a bunch of casting when using the DOM APIs to interact with the component class.